default.nix

```diff
@@ -26,68 +30,78 @@ let
     noTest = pkg: haskell.lib.dontCheck pkg;
 
     hspkgs = haskell.packages.${bootghc};
+    ghc    = haskell.compiler.${bootghc};
 
     ourtexlive =
       nixpkgs.texlive.combine {
         inherit (nixpkgs.texlive) scheme-small collection-xetex fncychap titlesec tabulary varwidth framed capt-of wrapfig needspace dejavu-otf helvetic; };
     fonts = nixpkgs.makeFontsConf { fontDirectories = [ nixpkgs.dejavu_fonts ]; };
     docsPackages = if withDocs then [ python3Packages.sphinx ourtexlive ] else [];
 
-    deps =
+    depsSystem = with stdenv.lib; (
       [ autoconf automake m4
         gmp.dev gmp.out glibcLocales
         ncurses.dev ncurses.out
         perl git file which python3
-        (hspkgs.ghcWithPackages (ps: [ ps.alex ps.happy ]))
         xlibs.lndir  # for source distribution generation
-        cabal-install
         zlib.out
         zlib.dev
       ]
       ++ docsPackages
-      ++ stdenv.lib.optional withLlvm llvm_6
-      ++ stdenv.lib.optional withNuma numactl
-      ++ stdenv.lib.optional withDwarf elfutils
-      ++ stdenv.lib.optional (! stdenv.isDarwin) pxz ;
+      ++ optional withLlvm llvm_7
+      ++ optional withNuma numactl
+      ++ optional withDwarf elfutils
+      ++ (if (! stdenv.isDarwin)
+          then [ pxz ]
+          else [
+            libiconv
+            darwin.libobjc
+            darwin.apple_sdk.frameworks.Foundation
+          ])
+    );
+
+    env = ghc;
 
-    env = buildEnv {
-      name = "ghc-build-environment";
-      paths = deps;
-    };
+    hadrianCabalExists = builtins.pathExists hadrianCabal;
+    hsdrv = if (builtins.trace "checking if ${toString hadrianCabal} is present:  ${if hadrianCabalExists then "yes" else "no"}"
```

Comment:
This is not very handy for me. If I'm understanding this correctly, we'll be building hadrian as part of the derivation most of/all the time. But I'm working on hadrian 
*a lot* and this is going to trigger a lot more work that I'd like, as I keep changing hadrian. Could we have a boolean arg `withHadrian` or something along those lines, to decide whether to include hadrian or not?

I'm also not entirely clear on why we either do `callCabal2nix` on hadrian or we return the custom derivation.

Reply:
@alpmestan, the intention is not to build `hadrian`, no!

The only intended effect of the whole thing is choice of where to source the set of `hadrian`'s dependencies from:
1. hard-coded knowledge in `depsHaskell` (which used to be the only option), or
2. getting it from the actual `hadrian.cabal`

Reply:
..and to clarify -- the second option is only activated if you pass `--argstr hadrianCabal /path/to/ghc/hadrian/hadrian.cabal`.